### PR TITLE
Replace color spectrum with react-colorful

### PR DESCRIPTION
**Pull request by nathanbrud**
## What this PR does
This PR replaces the color spectrum Iris implementation with [react-colorful package](https://omgovich.github.io/react-colorful/).

The current color select is very laggy and buggy.
One of the biggest bugs is that dragging stops when cursor is outside the spectrum so users can't select white / black which is very basic.

## Screenshots & Recordings 
https://user-images.githubusercontent.com/9106375/221582453-6f0a9a38-90ca-4946-9f81-c1effe52659d.mp4

## How it does that
A lot of deleted code from the ColorSpectrum component and replaced with react-colorful.
Some unused props deleted as well from the ColorSelect state.
**No breaking changes at all - all props work the same**

## Testing
Run storybook and see all variants ColorSelect can be used in.

### CODE REVIEW

**Comment by juliewongbandue** [comment]
Hey @nathanbrud, thanks for taking this on. Before we make such a large change, we should definitely add a test here.

**Comment by nathanbrud** [comment]
@juliewongbandue Sure, added test 💪

**Review by juliewongbandue — CHANGES_REQUESTED** [review]
Hey Nathan! Thanks (again) for taking this on. A few other packages in our Vimeo repo use some of the props removed in this PR, so to avoid introducing breaking changes in those places, a solution would be to make this a `ColorSelectV2` component.

Since we did not have tests for the original component, it's hard to know if we would be breaking anything. Moving forward, to refactor existing components, we should include tests for the original component first (with 100% coverage) in a separate PR. Then, changes can be made.

I just have some suggestions on the test for _this_ approach to the component though!

cc: @tylerthegrove

File: src/components/inputs/ColorSelect/ColorSelect.test.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  ui: ReactElement,
+  options?: Omit<RenderOptions, 'wrapper'>
+) => render(ui, { wrapper: Provider, ...options });
+
+const TEST_COLOR = '#CCCCCC';
+
+describe('ColorSelect', () => {
+  it('Renders ColorSelect input', () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
```
[the remark this answers is not in the document]
Comment (nathanbrud) [reply src/components/inputs/ColorSelect/ColorSelect.test.tsx:29]: Sure, changed to use aria-label

File: src/components/inputs/ColorSelect/ColorSelectInput.tsx [hunk 2]
```diff
@@ -45,6 +45,7 @@ export function ColorSelectInput({
         style={style}
         type="text"
         value={HEX}
+        data-testid="color-select-input"
```
[the remark this answers is not in the document]
Comment (nathanbrud) [reply src/components/inputs/ColorSelect/ColorSelectInput.tsx:48]: Sure, changed to use aria-label

File: src/components/inputs/ColorSelect/ColorSelect.test.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      'color-select-selected-value'
+    );
+    const selectedColorValue = selectedColor.getAttribute('color');
+    expect(selectedColorValue).toBe(TEST_COLOR);
+  });
+
+  it('Resets ColorSelect color using reset button', async () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
+    fireEvent.click(input);
```
[the remark this answers is not in the document]
Comment (nathanbrud) [reply src/components/inputs/ColorSelect/ColorSelect.test.tsx:82]: Change all clicks to `userEvent`

File: src/components/inputs/ColorSelect/ColorSelect.test.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  it('Renders ColorSelect input', () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
+    expect(input).toBeInTheDocument();
+  });
+
+  it('Renders ColorSelect picker using input', () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByTestId('color-select-input');
+    fireEvent.click(input);
```
[the remark this answers is not in the document]
Comment (nathanbrud) [reply src/components/inputs/ColorSelect/ColorSelect.test.tsx:36]: Change all clicks to `userEvent`

**Comment by nathanbrud** [comment]
Hi @juliewongbandue, I changed the clicks to `userEvent` and removed all `data-testId`.

One of the benefits of using a third party here is that they write most of the tests for us - [as you can see here](https://github.com/omgovich/react-colorful/blob/master/tests/components.test.js)
And we can cover only the delta between the library and our component.

Regarding creating a new v2 component - I can do it if you like but I think we're missing the point here, the current component is very bad - users cannot select black/white in many places across Vimeo because dragging doesn't work when moving the mouse outside the spectrum.
So overriding the current component will fix a critical bug.

Your call :)

**Pull request closed without merging**